src/kanata/cmd.rs

```diff
@@ -15,25 +15,25 @@ pub(super) fn run_cmd_in_thread(cmd_and_args: Vec<String>) -> std::thread::JoinH
         for arg in args {
             cmd.arg(arg);
         }
+        let cmd_str = format!(
+            "Program: {:?}, Arguments: {}",
+            cmd.get_program(),
+            cmd.get_args()
+                .map(|arg| format!("{:?}", arg))
```

Comment:
Looking at this code, it seems like this would be more efficient with the original `fold` implementation, but instead of using `format!`, `cmd.push(format!("{arg:?}"));`, finding some way to use `write!`, so that an intermediate `String` isn't constructed would be better.

Reply:
What about 6198bd5? Does it look good enough now?

Reply:
It still had intermediate allocations